src/api/controllers/channels.ts

```diff
@@ -64,3 +66,23 @@ export async function handleInternalChannelUpdate(data: DiscordPayload) {
 
   eventHandlers.channelUpdate?.(channel, cachedChannel);
 }
+
+export async function handleInternalInviteCreate(data: DiscordPayload) {
+  if (data.t !== "INVITE_CREATE") return;
+
+  const payload = data.d as InviteCreateEvent;
+  const channel = await cacheHandlers.get("channels", payload.channnel_id);
+  if (!channel) return;
+
+  eventHandlers.inviteCreate?.(payload, channel);
```

Comment:
We should not dispatch the snake-cased payload to the user... first we need to camelize the payload and then pass it to the listener.